File: front-end/Norlan-Svelte-TypeScript-master/src/lib/models/AssegnazioneDPI.ts
import { TipoDPI } from './Enums';
/*
Modello per la gestione delle assegnazioni dei Dispositivi di Protezione Individuale (DPI).
Traccia le consegne ai dipendenti, le relative scadenze e lo stato delle revisioni.
*/
export interface AssegnazioneDPIData {
	idAssegnazione: number;
	idDipendente: number;
	tipo: TipoDPI;
	dataConsegna: string;
	dataScadenzaRevisione: string;
	daRevisionare: boolean;
	nomeDpi?: string;
}

export class AssegnazioneDPI {
	idAssegnazione: number;
	idDipendente: number;
	tipo: TipoDPI;
	dataConsegna: string;
	dataScadenzaRevisione: string;
	daRevisionare: boolean;
	nomeDpi?: string;

	constructor(data: AssegnazioneDPIData) {
		this.idAssegnazione = data.idAssegnazione;
		this.idDipendente = data.idDipendente;
		this.tipo = data.tipo;
		this.dataConsegna = data.dataConsegna;
		this.dataScadenzaRevisione = data.dataScadenzaRevisione;
		this.daRevisionare = data.daRevisionare;
		this.nomeDpi = data.nomeDpi;
	}
}

File: back-end/norlan-client-portal-master/src/main/java/it/norlan/clientportal/model/AssegnazioneDPI.java
package it.norlan.clientportal.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

/**
 * Entità JPA per il tracciamento dei Dispositivi di Protezione Individuale (DPI).
 * Memorizza la consegna logistica ai dipendenti e le date limite per la
 * revisione obbligatoria o la sostituzione dei presidi di sicurezza.
 */

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Entity
@Table(name = "assegnazione_dpi")
public class AssegnazioneDPI {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idAssegnazione;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_dipendente", nullable = false)
    private Dipendente dipendente;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoDPI tipo;

    @Column(name = "data_consegna")
    private LocalDate dataConsegna = LocalDate.now();

    @Column(name = "data_scadenza_revisione", nullable = false)
    private LocalDate dataScadenzaRevisione;

    public enum TipoDPI { ELMETTO, GUANTI, SCARPE_ANTINFORTUNISTICHE, OCCHIALI, ALTRO }

    @Column(name = "nome_dpi_personalizzato")
    private String nomeDpi;
}
